### DISCUSSION TITLE
what is bloom filter and what it does?

### DISCUSSION THREAD

**Discussion by fre2mansur** [albertobsd/keyhunt#334] [General]
1. I was wondering what is bloom filter it showed me bloom filter: 84.

2. also it says by default -R is random, but I choose -B Sequential, which one will work? 
3. and what is the backward how it works?

**Comment by albertobsd** [comment]
1. 84? Where what are you doing, can you detail it? Bloom filter is a data structure to discard data that is not in your "database" without perform a full deep search row by row.
2. if you ommit the -R it will be automatically sequential without need of -B something. the -B option only works for BSGS
3. Is not obvious? It search the range backwards, only works for BSGS

**Reply by fre2mansur** [reply]
here the command i run 
`./keyhunt -m rmd160 -f tests/unsolvedpuzzles.rmd -b 66 -l compress -B backward -q  -t 2`

I received this
```
Bloom filter for 84 elements.
[+] Loading data to the bloomfilter total: 0.03 MB
[+] Sorting data ... done! 84 values were loaded and sorted
```

Now I'm wondering what is BSGS,

"Bloom filter is a data structure to discard data that is not in your "database" without perform a full deep search row by row."
Do i need to have database?

**Reply by albertobsd** [reply]
No you don't need a database. for database i mean the Data set of target keys that you give to the program in the input file `-f tests/unsolvedpuzzles.rmd`

**Reply by fre2mansur** [reply]
So will the program find the private key for all or any given items in the unsolved puzzle?
Do you think my command is efficient, by chance? 
./keyhunt -m rmd160 -f tests/unsolvedpuzzles.rmd -b 66 -l compress -B backward -q  -t 2 ?
